server/src/com/thoughtworks/go/server/service/EnvironmentConfigService.java

```diff
@@ -145,6 +136,21 @@ public EnvironmentConfig named(String environmentName) throws NoSuchEnvironmentE
         return environments.named(new CaseInsensitiveString(environmentName));
     }
 
+    public EnvironmentConfig getEnvironmentConfig(String environmentName) throws NoSuchEnvironmentException {
+        return environments.named(new CaseInsensitiveString(environmentName));
+    }
+
+    public void deleteEnvironment(String environmentName, HttpLocalizedOperationResult result) throws NoSuchEnvironmentException {
+        if (environments.hasEnvironmentNamed(new CaseInsensitiveString(environmentName))) {
+            CruiseConfig config = goConfigService.getConfigForEditing();
```

Comment:
You should never edit `goConfigService.getConfigForEditing()` directly. You would normally make all relevant changes to a copy of config and then send it for save. Once all the validations pass and save is completed, `getConfigForEditing` on the service would start returning the updated config. Look at `PipelineConfigService` ([this](https://github.com/gocd/gocd/blob/5f54cf4338eb2ff5c9f486ede6ae97e9aae51e82/server/src/com/thoughtworks/go/server/service/PipelineConfigService.java#L97) and [this](https://github.com/gocd/gocd/blob/5f54cf4338eb2ff5c9f486ede6ae97e9aae51e82/server/src/com/thoughtworks/go/server/service/PipelineConfigService.java#L134)) to know more about what I mean